shared/feed-parser.js

```diff
@@ -6,6 +6,7 @@ const FeedParser = {
     return new Promise((resolve, reject) => {
       const request = new XMLHttpRequest();
       request.open("GET", url, true);
+      request.timeout = 10000; // time in milliseconds
```

Comment:
Can we make this 5 seconds ?

```suggestion
      request.timeout = 5000; // time in milliseconds
```